src/rt/cpp/when.h

```diff
@@ -14,13 +14,102 @@ namespace verona::cpp
 {
   using namespace verona::rt;
 
+  template<typename T>
+  struct cown_ptr_span
+  {
+    cown_ptr<T>* array;
+    size_t length;
+
+    void constr_helper(cown_ptr<T>* arr)
+    {
+      array = reinterpret_cast<cown_ptr<T>*>(
+        snmalloc::ThreadAlloc::get().alloc(length * sizeof(cown_ptr<T>*)));
+      memset(array, 0, length * sizeof(cown_ptr<T>*));

```

Comment:
Why memset? The next loop overwrites everything.

Reply:
Because the `cown_ptr` copy assignment operator does a `clear()` first. I'll remove the memset and call the copy constructor with `new (&array[i]) cown_ptr<T>(arr[i])` instead.